### PR TITLE
YC - first week

### PR DESCRIPTION
**Pull request by VCneverdie**
- 4 classes and corresponding tests: Block(Italic, Link, Strikethrough) and Paragraph(Unordered list)
- I pulled Andy's base class for Block and Paragraph so that I put all classes inside `element.py`. Personally I do want them in separate files. At least `element_block.py` and `element_paragraph.py`

### CODE REVIEW

File: element_test.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    ( "", "<em></em>")
+])
+def test_italic_render(content, html):
+    ib = ItalicBlock(content)
+    assert html == ib.render()
+
+@pytest.mark.parametrize("content, isStart, isEnd, html", [
+    ( "first item", True, None, "<ul><li>first item</li>" ),
+    ( "second item", None, None, "<li>second item</li>" ),
+    ( "last item", None, True, "<li>last item</li></ul>" )
```
Comment (VCneverdie) [inline element_test.py]: Should we consider the nested case for a paragraph element like an unordered list?

Comment (andysim3d) [reply element_test.py]: we should support both ordered list and unordered list.

Comment (jinglongwu) [reply element_test.py:35]: > we should support both ordered list and unordered list.

that's my part

Comment (jinglongwu) [reply element_test.py]: right, even though ordered list is my part, but these two styles are highly similar. Let's rename the class to `List` and add "isOrdered" flag to indicate it's ordered or unordered. Then, the start or end could be determined correspondingly

**Review by andysim3d — CHANGES_REQUESTED** [review]
Please sync and update

**Review by andysim3d — APPROVED** [review]

**Pull request merged**